File: plugins/java/old/Annihilation - minetox/src/Annihilation/Arena/Manager/DeathManager.java
package Annihilation.Arena.Manager;

import Annihilation.Annihilation;
import Annihilation.Arena.Arena;
import Annihilation.Arena.Kits.Kit;
import Annihilation.Arena.Object.PlayerData;
import Annihilation.MySQL.NormalQuery;
import cn.nukkit.entity.Entity;
import cn.nukkit.entity.projectile.EntityProjectile;
import cn.nukkit.event.player.PlayerDeathEvent;
import cn.nukkit.event.entity.EntityDamageEvent;
import cn.nukkit.event.entity.EntityDamageByEntityEvent;
import cn.nukkit.event.entity.EntityDamageByChildEntityEvent;
import cn.nukkit.Player;
import cn.nukkit.utils.TextFormat;

public class DeathManager {

    public Arena plugin;

    public DeathManager(Arena plugin) {
        this.plugin = plugin;
    }

    public void onDeath(PlayerDeathEvent e) {
        Player p = e.getEntity();
        EntityDamageEvent lastDmg = p.getLastDamageCause();
        String pColor = this.plugin.getPlayerTeam(p).getColor();
        String dColor = "";
        boolean escape = false;
        if (lastDmg != null) {
            if (lastDmg instanceof EntityDamageByChildEntityEvent) {
                Entity arrow = ((EntityDamageByChildEntityEvent) lastDmg).getChild();
                Entity killer = ((EntityDamageByChildEntityEvent) lastDmg).getDamager();
                if (arrow instanceof EntityProjectile) {
                    this.plugin.messageAllPlayers(pColor + p.getName() + TextFormat.GRAY + " was shot by " + this.plugin.getPlayerTeam((Player) killer).getColor() + killer.getName());
                    new NormalQuery(Annihilation.getInstance(), NormalQuery.KILL, new String[]{killer.getName().toLowerCase()});
                    new NormalQuery(Annihilation.getInstance(), "tokens", new String[]{killer.getName().toLowerCase()}, 10, "freezecraft");
                }
                return;
            } else if (lastDmg instanceof EntityDamageByEntityEvent) {
                Entity killer = ((EntityDamageByEntityEvent) lastDmg).getDamager();
                if (killer instanceof Player) {
                    dColor = this.plugin.getPlayerTeam((Player) killer).getColor();
                    this.plugin.messageAllPlayers(pColor + p.getName() + TextFormat.GRAY + " was slain by " + dColor + killer.getName());
                    new NormalQuery(Annihilation.getInstance(), NormalQuery.KILL, new String[]{killer.getName().toLowerCase()});
                    new NormalQuery(Annihilation.getInstance(), "tokens", new String[]{killer.getName().toLowerCase()}, 10, "freezecraft");
                }
                return;
            }

            String killer = null;

            PlayerData killerData = this.plugin.getPlayerData(p).wasKilled();

            if (killerData != null) {
                escape = true;
                dColor = killerData.getTeam().getColor();
                killer = killerData.getName();
            }
            /*if($lastDmg instanceof EntityDamageByBlockEvent){
                if($escape === true){
                    $this->plugin->messageAllPlayers($pColor."{$p->getName()}".TextFormat::GRAY." walked into a cactus while trying to escape ".$this->plugi->getTeamColor($this->plugin->getPlayerTeam($killer)).$killer->getName());
                    $this->plugin->mysql->addKill($killer->getName());
                    return;
                }
                $this->plugin->messageAllPlayers($pColor."{$p->getName()}".TextFormat::GRAY." was pricked to death");
                return;
            }*/
            if (escape) {
                new NormalQuery(Annihilation.getInstance(), NormalQuery.KILL, new String[]{killer.toLowerCase()});
                new NormalQuery(Annihilation.getInstance(), "tokens", new String[]{killer.toLowerCase()}, 10, "freezecraft");
                Player pl = this.plugin.plugin.getServer().getPlayer(killer);
                if (pl != null) {
                    pl.addExperience(15);
                    if (this.plugin.getPlayerData(pl).getKit() == Kit.BERSERKER && pl.getMaxHealth() < 30) {
                        pl.setMaxHealth(pl.getMaxHealth() + 1);
                    }
                }
            }

            switch (lastDmg.getCause()) {
                case 0:
                    if (escape) {
                        this.plugin.messageAllPlayers(pColor + p.getName() + TextFormat.GRAY + " walked into a cactus while trying to escape " + dColor + killer);
                        return;
                    }
                    this.plugin.messageAllPlayers(pColor + p.getName() + TextFormat.GRAY + " was pricked to death");
                    break;
                case 3:
                    this.plugin.messageAllPlayers(pColor + p.getName() + TextFormat.GRAY + " suffocated in a wall");
                    break;
                case 4:
                    if (escape) {
                        this.plugin.messageAllPlayers(pColor + p.getName() + TextFormat.GRAY + " was doomed to fall by " + dColor + killer);
                        break;
                    }
                    this.plugin.messageAllPlayers(pColor + p.getName() + TextFormat.GRAY + " fell from high place");
                    break;
                case 5:
                    if (escape) {
                        this.plugin.messageAllPlayers(pColor + p.getName() + TextFormat.GRAY + " walked into a fire whilst fighting " + dColor + killer);
                        break;
                    }
                    this.plugin.messageAllPlayers(pColor + p.getName() + TextFormat.GRAY + " went up in flames");
                    break;
                case 6:
                    if (escape) {
                        this.plugin.messageAllPlayers(pColor + p.getName() + TextFormat.GRAY + " was burnt to a crisp whilst fighting " + dColor + killer);
                        break;
                    }
                    this.plugin.messageAllPlayers(pColor + p.getName() + TextFormat.GRAY + " burned to death");
                    break;
                case 7:
                    if (escape) {
                        this.plugin.messageAllPlayers(pColor + p.getName() + TextFormat.GRAY + " tried to swim in lava while trying to escape " + dColor + killer);
                        break;
                    }
                    this.plugin.messageAllPlayers(pColor + p.getName() + TextFormat.GRAY + " tried to swim in lava");
                    break;
                case 8:
                    if (escape) {
                        this.plugin.messageAllPlayers(pColor + p.getName() + TextFormat.GRAY + " drowned whilst trying to escape " + dColor + killer);
                        break;
                    }
                    this.plugin.messageAllPlayers(pColor + p.getName() + TextFormat.GRAY + " drowned");
                    break;
                case 9:
                    this.plugin.messageAllPlayers(pColor + p.getName() + TextFormat.GRAY + " blew up");
                    break;
                case 10:
                    this.plugin.messageAllPlayers(pColor + p.getName() + TextFormat.GRAY + " blew up");
                    break;
                case 11:
                    if (escape) {
                        this.plugin.messageAllPlayers(pColor + p.getName() + TextFormat.GRAY + " was doomed to fall by " + dColor + killer);
                        break;
                    }
                    this.plugin.messageAllPlayers(pColor + p.getName() + TextFormat.GRAY + " fell out of the world");
                    break;
                case 12:
                    this.plugin.messageAllPlayers(pColor + p.getName() + TextFormat.GRAY + " died");
                    break;
                case 13:
                    this.plugin.messageAllPlayers(pColor + p.getName() + TextFormat.GRAY + " died");
                    break;
                case 14:
                    this.plugin.messageAllPlayers(pColor + p.getName() + TextFormat.GRAY + " died");
                    break;
            }
        }
    }
}

File: plugins/NoveBedWars/manager/DeathManager.php
<?php

namespace BedWars\manager;

use BedWars\arena\Arena;
use BedWars\game\GamePlayer;
use pocketmine\event\player\PlayerDeathEvent;
use pocketmine\event\entity\EntityDamageEvent;
use pocketmine\event\entity\EntityDamageByEntityEvent;
use pocketmine\event\entity\EntityDamageByChildEntityEvent;
use pocketmine\entity\Projectile;
use pocketmine\Player;
use pocketmine\utils\TextFormat;

class DeathManager{
    
    public $arena;
    
    public function __construct(Arena $arena){
        $this->arena = $arena;
    }
    
    public function onDeath(PlayerDeathEvent $e){
        /** @var Player $p */
        $p = $e->getEntity();
        $game = $this->arena->getPlayer($p) instanceof GamePlayer ? $this->arena->getPlayer($p) : null;
        $lastDmg = $p->getLastDamageCause();
        $pColor = $game->getTeam()->getChat();
        $escape = false;
        if($lastDmg instanceof EntityDamageEvent){
            if($lastDmg instanceof EntityDamageByEntityEvent){
                /** @var Player $killer */
                $killer = $lastDmg->getDamager();
                $gamer = $this->arena->getPlayer($killer) instanceof GamePlayer ? $this->arena->getPlayer($killer) : null;
                if($killer instanceof Player and $gamer instanceof GamePlayer){
                    $dColor = $gamer->getTeam()->getChat();
                    $this->arena->messageAllPlayers($pColor."{$p->getName()}".TextFormat::GRAY." was slain by ".$dColor."{$killer->getName()}");
                    $this->arena->bw->mysqlmgr->addKill($killer->getName());
                }
                return;
            }
            if($lastDmg instanceof EntityDamageByChildEntityEvent){
                $arrow = $lastDmg->getChild();
                /** @var Player $killer */
                $killer = $lastDmg->getDamager();
                if($arrow instanceof Projectile){
                    $gamer = $this->arena->getPlayer($killer) instanceof GamePlayer ? $this->arena->getPlayer($killer) : null;
                    $this->arena->messageAllPlayers($pColor."{$p->getName()}".TextFormat::GRAY." was shot by ".$gamer->getTeam()->getChat().$killer->getName());
                    $this->arena->bw->mysqlmgr->addKill($killer->getName());
                }
                return;
            }
            $killer = null;
            if(isset($this->arena->bw->listener->players[strtolower($p->getName())]['tick']) && $this->arena->bw->listener->players[strtolower($p->getName())]['tick'] >= $this->arena->bw->getServer()->getTick()){
                $escape = true;
                $dColor = $this->arena->bw->listener->players[strtolower($p->getName())]['killer_color'];
                $killer = $this->arena->bw->listener->players[strtolower($p->getName())]['killer'];
            }
        switch($lastDmg->getCause()){
            case 0:
                if($escape === true){
                    $this->arena->messageAllPlayers($pColor."{$p->getName()}".TextFormat::GRAY." walked into a cactus while trying to escape ".$dColor.$killer);
                    $this->arena->bw->mysqlmgr->addKill($killer);
                    return;
                }
                $this->arena->messageAllPlayers($pColor."{$p->getName()}".TextFormat::GRAY." was pricked to death");
                break;
            case 3:
                if($escape === true){
                    $this->arena->bw->mysqlmgr->addKill($killer);
                }
                $this->arena->messageAllPlayers($pColor."{$p->getName()}".TextFormat::GRAY." suffocated in a wall");
                break;
            case 4:
                if($escape === true){
                    $this->arena->messageAllPlayers($pColor."{$p->getName()}".TextFormat::GRAY." was doomed to fall by ".$dColor.$killer);
                    $this->arena->bw->mysqlmgr->addKill($killer);
                    break;
                }
                $this->arena->messageAllPlayers($pColor."{$p->getName()}".TextFormat::GRAY." fell from high place");
                break;
            case 5:
                if($escape === true){
                    $this->arena->messageAllPlayers($pColor."{$p->getName()}".TextFormat::GRAY." walked into a fire whilst fighting ".$dColor.$killer);
                    $this->arena->bw->mysqlmgr->addKill($killer);
                    break;
                }
                $this->arena->messageAllPlayers($pColor."{$p->getName()}".TextFormat::GRAY." went up in flames");
                break;
            case 6:
                if($escape === true){
                    $this->arena->messageAllPlayers($pColor."{$p->getName()}".TextFormat::GRAY." was burnt to a crisp whilst fighting ".$dColor.$killer);
                    $this->arena->bw->mysqlmgr->addKill($killer);
                    break;
                }
                $this->arena->messageAllPlayers($pColor."{$p->getName()}".TextFormat::GRAY." burned to death");
                break;
            case 7:
                if($escape === true){
                    $this->arena->messageAllPlayers($pColor."{$p->getName()}".TextFormat::GRAY." tried to swim in lava while trying to escape ".$dColor.$killer);
                    $this->arena->bw->mysqlmgr->addKill($killer);
                    break;
                }
                $this->arena->messageAllPlayers($pColor."{$p->getName()}".TextFormat::GRAY." tried to swim in lava");
                break;
            case 8:
                if($escape === true){
                    $this->arena->messageAllPlayers($pColor."{$p->getName()}".TextFormat::GRAY." drowned whilst trying to escape ".$dColor.$killer);
                    $this->arena->bw->mysqlmgr->addKill($killer);
                    break;
                }
                $this->arena->messageAllPlayers($pColor."{$p->getName()}".TextFormat::GRAY." drowned");
                break;
            case 9:
                if($escape === true){
                    $this->arena->bw->mysqlmgr->addKill($killer);
                }
                $this->arena->messageAllPlayers($pColor."{$p->getName()}".TextFormat::GRAY." blew up");
                break;
            case 10:
                if($escape === true){
                    $this->arena->bw->mysqlmgr->addKill($killer);
                }
                $this->arena->messageAllPlayers($pColor."{$p->getName()}".TextFormat::GRAY." blew up");
                break;
            case 11:
                if($escape === true){
                    $this->arena->messageAllPlayers($pColor."{$p->getName()}".TextFormat::GRAY." was doomed to fall by ".$dColor.$killer);
                    $this->arena->bw->mysqlmgr->addKill($killer);
                    break;
                }
                $this->arena->messageAllPlayers($pColor."{$p->getName()}".TextFormat::GRAY." fell out of the world");
                break;
            case 12:
                if($escape === true){
                    $this->arena->bw->mysqlmgr->addKill($killer);
                }
                $this->arena->messageAllPlayers($pColor."{$p->getName()}".TextFormat::GRAY." died");
                break;
            case 13:
                if($escape === true){
                    $this->arena->bw->mysqlmgr->addKill($killer);
                }
                $this->arena->messageAllPlayers($pColor."{$p->getName()}".TextFormat::GRAY." died");
                break;
            case 14:
                if($escape === true){
                    $this->arena->bw->mysqlmgr->addKill($killer);
                }
                $this->arena->messageAllPlayers($pColor."{$p->getName()}".TextFormat::GRAY." died");
                break;
        }
        }
    }
}